new scope guidance (#69)
ubclaunchpad/docs: handbook/project-management/scope.md

# 🔍 Project Scope <Badge type="tip" text="updated"/>

This document outlines advice on how to scope your project in a way that makes it easier to deliver on the (comparatively) tight schedules Launch Pad projects have.

## Launch Pad Limitations

> Note that on paper, it may seem like you have a lot of time since teams get a whole semester to complete their project. However, in between designing your project, learning the prerequisite technologies, adapting to feedback, midterms, breaks, absences, and more, teams don't actually have a lot of time to complete projects.

Reviewer: Note that on paper, it may seem like you have a lot of time since teams get a whole semester to complete their project. However, in between designing your project, learning the prerequisite technologies, adapting to feedback, midterms, breaks, absences, and more, many previous teams actually struggled to complete the projects.

## Ideation

### Core Feature and the MVP

In general, your idea should be able to be distilled down to a single core feature. For example:

| Idea | Core Feature |
|------|--------------|
| "a website where you can watch YouTube Videos with friends and chat and queue up playlists" | ability to watch YouTube videos in sync
| "a choose-your-own adventure game where you try and survive as a UBC student" | interface to make choices and present consequences
| "slack bot and team management system for Launch Pad" | update GitHub teams via Slack

This should be done *before* you start any sort of development. Using this core feature, you should determine:

* 💪 **Requirements**: Everything you need to achieve to deliver your core feature, and *nothing else*.
* 🛣 **Stretch Goals**: A *small* set of additional things that must be done to deliver on anything you want to achieve that does not fall within your core feature.

We call the core feature and this set of "requirements" the **Minimum Viable Product**, or MVP.

### Project Timeline

::: warning
This timeline is a *new and proposed* timeline, and will probably change based on our experiences trying it out.
:::

> * **Week 0**: Brain storm ideas and start to consolidate around one or two

Reviewer: Brainstorm (one word)

We call the core feature and this set of "requirements" the **Minimum Viable Product**, or MVP.

### Project Timeline

::: warning
This timeline is a *new and proposed* timeline, and will probably change based on our experiences trying it out.
:::

* **Week 0**: Brain storm ideas and start to consolidate around one or two
  * Designers will focus on user testing and prototyping
* **Week 2**: Finalized project pitch and MVP writeup
  * Designers will continue user testing and prototyping with a focus on the MVP
  * Developers will start implementation work
* **Week 5**: A *functional* MVP based on initial idea and prototypes
  * Teams should start asking people to try initial implementation and give ideas and feedback
  * Designers should start work on collecting feedback based on the implementation, branding, and improving the MVP
  * Developers should work on making the core feature work well and implementing designs
  * Remarks:
    * At this point, the MVP will be quite clunky but should capture the MVP's core feature
    * At this point, midterms and whatnot will probably start to take a toll on progress

> * **Week 8**: Branding/design starts to come together, MVP looks nicer and smoother, some progress on stretch goals might be made

Reviewer 2: Wording is a bit generous. I think the expectations can be firmer here.
```suggestion
* **Week 8**: Branding/design starts to come together, MVP is more polished, progress is made on stretch goals
```

  * Teams should continue collecting feedback based on their in-progress implementation
  * Designers should start putting together promotional material and helping developers finalize design implementation

>   * Developers should shift focus to polish and a few stretch goals

Reviewer 2: ```suggestion
  * Developers should shift focus to polish and stretch goals
```
